### PR TITLE
Fixes #23750 - add toast notification when a task is started

### PR DESCRIPTION
**Pull request by amirfefer**
When a task is started in the subscriptions page, a toast notification will pop out

192 168 121 17_3000_xui_subscriptions 2

### CODE REVIEW

File: webpack/scenes/Subscriptions/SubscriptionsPage.js [hunk 1]
```diff
@@ -60,6 +60,12 @@ class SubscriptionsPage extends Component {
     if (numberOfTasks > 0) {
       if (numberOfPrevTasks === 0 || prevProps.tasks[0].id !== tasks[0].id) {
         [task] = this.props.tasks;
+        const taskStatedMessage = Jed.sprintf(__('%s task has started.'), task.humanized.action);
```
Comment (waldenraines) [inline webpack/scenes/Subscriptions/SubscriptionsPage.js]: Is it possible to include a link to the task like we do with the other toast notifications?

Comment (amirfefer) [reply webpack/scenes/Subscriptions/SubscriptionsPage.js]: sure

Comment (tstrachota) [reply webpack/scenes/Subscriptions/SubscriptionsPage.js]: There's a typo in the variable name: 
```
taskStartedMessage
       ^
```

**Comment by tstrachota** [comment]
Not sure how big deal it is, but with this approach the notification appears also on each page reload. For a long running tasks it tells you that the task has started when it's already in the middle of its run.

**Comment by waldenraines** [comment]
> Not sure how big deal it is, but with this approach the notification appears also on each page reload. For a long running tasks it tells you that the task has started when it's already in the middle of its run.

Hmm, yeah, I wonder if we could just reword it to "your task is running...".  @rohoover, any thoughts on this?

**Comment by johnpmitsch** [comment]
You could use `localStorage` to persist if a notification has been sent on refresh. This seems to work for me, but please test further if you would like to use this approach (also feel free to refactor/clean up)

```diff
diff --git a/webpack/scenes/Subscriptions/SubscriptionsPage.js b/webpack/scenes/Subscriptions/SubscriptionsPage.js
index b459a49..49b9dd6 100644
--- a/webpack/scenes/Subscriptions/SubscriptionsPage.js
+++ b/webpack/scenes/Subscriptions/SubscriptionsPage.js
@@ -61,17 +61,23 @@ class SubscriptionsPage extends Component {
     if (numberOfTasks > 0) {
       if (numberOfPrevTasks === 0 || prevProps.tasks[0].id !== tasks[0].id) {
         [task] = this.props.tasks;
-        const taskStatedMessage = Jed.sprintf(__('%s task has started.'), task.humanized.action);
-
-        notify({
-          message: ReactDOMServer.renderToStaticMarkup(taskStatedMessage),
-          type: 'success',
-        });
+        const notifiedTasks = localStorage.getItem("notifiedTasks") || [];
+        if (!notifiedTasks.includes(task.id)) {
+          const taskStatedMessage = Jed.sprintf(__('%s task has started.'), task.humanized.action);
+
+          notify({
+            message: ReactDOMServer.renderToStaticMarkup(taskStatedMessage),
+            type: 'success',
+          });
+          
+          localStorage.setItem("notifiedTasks", [...notifiedTasks, task.id]);
+        }
         this.handleDoneTask(task);
       }
     }
   }
 
+
```

**Comment by tstrachota** [comment]
Ain't using local storage too complicated? How about not sending the notification from `componentDidUpdate` but when the action is actually triggered?

**Comment by johnpmitsch** [comment]
@tbrisker its not conventional, but it does work :) I don't know how to persist state on a page reload any other way (outside of another dependency like [redux-persist](https://github.com/rt2zz/redux-persist)). Maybe I am misunderstanding the problem though.

Sending the notification when the action is triggered sounds fine, but will that work with the way we handle tasks? I thought we are using a polling method to the server to create a list of tasks? Won't it always pick up the manifest task on page load?

**Comment by tbrisker** [comment]
@johnpmitsch i think you wanted to mention @tstrachota  :)

**Comment by johnpmitsch** [comment]
@tbrisker sorry! :man_facepalming: @tstrachota see above :)

**Comment by tstrachota** [comment]
@johnpmitsch we handle tasks that way only for tracking their progress. If you trigger a task, lets say manifest upload, and server replies with 202 the it's imho safe to show the notification that the task has started. When a user reloads the page the tasks will load to show the progress but no notification will appear.

**Comment by johnpmitsch** [comment]
@tstrachota that sounds like a reasonable approach, thanks for clarifying :+1:

**Comment by amirfefer** [comment]
>  If you trigger a task, lets say manifest upload, and server replies with 202 the it's imho safe to show the notification that the task has started. 

@tstrachota afaik with this approach we have no task id, and we need it for url creation, though we can create a toast notification without a task link.

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 2]
```diff
@@ -114,12 +123,37 @@ class ManageManifestModal extends Component {
...
+      organization.owner_details && organization.owner_details.upstreamConsumer
+    );
+  }
+
+  renderTaskStartedToast() {
+    const { taskDetails } = this.props;
+    const message = (
+      <span>
+        <span>
+          {__(`Task ${taskDetails.humanized.action} has started`)}
```
Comment (waldenraines) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Missing punctuation here.  Maybe make this "Task ____ has started.  Click here to go to the tasks page for the task."

Comment (tstrachota) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Watch out, you need to use `sprintf` here.

**Comment by tstrachota** [comment]
> Also, do you mind adding Actions::Katello::UpstreamSubscriptions::UpdateEntitlements to the list of BLOCKING_FOREMAN_TASK_TYPES so the progress bar will be displayed when entitlements are edited? @tstrachota please sanity check me on this last request.

That makes sense.

**Comment by waldenraines** [comment]
@amirfefer this needs some updates please.

**Comment by waldenraines** [comment]
@amirfefer ping on the above.

**Comment by amirfefer** [comment]
@waldenraines,
it's rebased and updated.

**Comment by waldenraines** [comment]
@akofink there are a couple of test failures to fix here.

**Comment by johnpmitsch** [comment]
@waldenraines I think you mean @amirfefer  ;)

**Comment by waldenraines** [comment]
> @waldenraines I think you mean @amirfefer ;)

Hah, oops.

**Comment by waldenraines** [comment]
[test katello]

**Comment by waldenraines** [comment]
@johnpmitsch any other requested changes on this?
